app/frontend/wallet/shelley/shelley-trezor-crypto-provider.ts

```diff
@@ -401,11 +404,11 @@ const ShelleyTrezorCryptoProvider = async ({
         return {
           finalizedTxAux: ShelleyTxAux({
             ...txAux,
-            auxiliaryDataHash: auxiliaryDataSupplement.auxiliaryDataHash,
+            auxiliaryDataHash: auxiliaryDataSupplement?.auxiliaryDataHash,
```

Comment:
auxiliaryDataSupplement shouldn't really be undefined here, by the contract of the function called it should exist, so I'd rather add here a runtime assertion

Reply:
I am seeing in the docs that it can be undefined https://github.com/trezor/connect/blob/2955b5ef92a3c77ccf3b1353f74f37fa4a2ff337/docs/methods/cardanoSignTransaction.md#flowtype-1

I can still replace it with a runtime assertion

Reply:
yes, indeed the docs aren't very clear on that now that I checked it deeper, sorry for the confusion, but it's supposed to be defined anyway for catalyst registrations and if not, this code wouldn't work as we need these data in such case. So a runtime assertion makes sense